Gems/Blast/Editor/Scripts/blast_asset_builder.py

```diff
@@ -0,0 +1,290 @@
+
+    # create assetinfo generated file
+    productOutputs.append(generate_assetinfo_product(request))
+
+    response.outputProducts = productOutputs
+    response.resultCode = azlmbr.asset.builder.ProcessJobResponse_Success
+    response.dependenciesHandled = True
+    return response
+
+# using the incoming 'request' find the type of job via 'jobKey' to determine what to do
+def on_process_job(args):
+    try:
+        request = args[0]
+        if (request.jobDescription.jobKey.startswith(jobKeyName)):
+            return process_fbx_file(request)
+
+        return azlmbr.asset.builder.ProcessJobResponse()
+    except:
+        log_exception_traceback()
+        return azlmbr.asset.builder.ProcessJobResponse()
+
+# register asset builder
+def register_asset_builder():
+    assetPattern = azlmbr.asset.builder.AssetBuilderPattern()
+    assetPattern.pattern = '*.blast'
+    assetPattern.type = azlmbr.asset.builder.AssetBuilderPattern_Wildcard
+
+    builderDescriptor = azlmbr.asset.builder.AssetBuilderDesc()
+    builderDescriptor.name = "Blast Scene Builder"
+    builderDescriptor.patterns = [assetPattern]
+    builderDescriptor.busId = busId
+    builderDescriptor.version = 1
+
+    outcome = azlmbr.asset.builder.PythonAssetBuilderRequestBus(azlmbr.bus.Broadcast, 'RegisterAssetBuilder', builderDescriptor)
+    if outcome.IsSuccess():
+        # created the asset builder to hook into the notification bus
+        handler = azlmbr.asset.builder.PythonBuilderNotificationBusHandler()
+        handler.connect(busId)
+        handler.add_callback('OnCreateJobsRequest', on_create_jobs)
+        handler.add_callback('OnProcessJobRequest', on_process_job)
+        return handler
+
+# create the asset builder handler
+pythonAssetBuilderHandler = None
+try:
+    if (pythonAssetBuilderHandler == None):
+        pythonAssetBuilderHandler = register_asset_builder()
+except:
+    pythonAssetBuilderHandler = None
+
+#
+# SceneAPI Processor
+#
+blastChunksAssetType = azlmbr.math.Uuid_CreateString('{993F0B0F-37D9-48C6-9CC2-E27D3F3E343E}', 0)
+
+def export_chunk_asset(scene, outputDirectory, platformIdentifier, productList):
+    import azlmbr.scene
+    import azlmbr.object
+    import azlmbr.paths
+    import json, os
+
+    jsonFilename = os.path.basename(scene.sourceFilename)
+    jsonFilename = os.path.join(outputDirectory, jsonFilename + '.blast_chunks')
+
+    # prepare output folder
+    basePath, _ = os.path.split(jsonFilename)
+    outputPath = os.path.join(outputDirectory, basePath)
+    if not os.path.exists(outputPath):
+        os.makedirs(outputPath, False)
+
+    # write out a JSON file with the chunk file info
+    with open(jsonFilename, "w") as jsonFile:
+        jsonFile.write(scene.manifest.ExportToJson())
+
+    exportProduct = azlmbr.scene.ExportProduct()
+    exportProduct.filename = jsonFilename
+    exportProduct.sourceId = scene.sourceGuid
+    exportProduct.assetType = blastChunksAssetType
+    exportProduct.subId = 101
```

Comment:
is there any significance to this subId? Is there a place we can document this?

I've already had people ask me questions about figuring out which sub IDs for products from builders map to which assets, and there haven't been good patterns for people to follow besides "go read the logic for that builder that applies sub IDs, and hope it won't change on you later"

Reply:
yes, it needs to be different from anything that already comes from a FBX file and 101 looked safe. 

Sadly that is the way the sub-id system works right now since there is no other way to uniquely ID itself from other conflicts like AZ Model, material, or other things that the Atom team want to spit out

This is a weak link right now the over all asset pipeline.